File: student-management-frontend/src/services/AcademicService.js
// src/services/AcademicService.js
import axios from 'axios';
import { ACADEMIC_API_BASE } from './apiConfig';

// Academic Years
export const getAllYears = () => axios.get(`${ACADEMIC_API_BASE}/years`);
export const createYear = (year) => axios.post(`${ACADEMIC_API_BASE}/years`, year);

// Standards (Classes)
export const getAllStandards = () => axios.get(`${ACADEMIC_API_BASE}/standards`);
export const createStandard = (standard) => axios.post(`${ACADEMIC_API_BASE}/standards`, standard);

// Sections
export const getAllSections = () => axios.get(`${ACADEMIC_API_BASE}/sections`);
export const createSection = (section) => axios.post(`${ACADEMIC_API_BASE}/sections`, section);

// Subjects
export const getAllSubjects = () => axios.get(`${ACADEMIC_API_BASE}/subjects`);
export const createSubject = (subject) => axios.post(`${ACADEMIC_API_BASE}/subjects`, subject);

File: student-management-backend/src/main/java/com/example/sms/service/AcademicService.java
package com.example.sms.service;

import com.example.sms.entity.AcademicYear;
import com.example.sms.entity.Section;
import com.example.sms.entity.Standard;
import com.example.sms.entity.Subject;

import java.util.List;

public interface AcademicService {

    // Academic Year
    AcademicYear createAcademicYear(AcademicYear academicYear);

    List<AcademicYear> getAllAcademicYears();

    AcademicYear getAcademicYearById(Long id);

    // Standard (Class)
    Standard createStandard(Standard standard);

    List<Standard> getAllStandards();

    Standard getStandardById(Long id);

    // Section
    Section createSection(Section section);

    List<Section> getAllSections();

    Section getSectionById(Long id);

    // Subject
    Subject createSubject(Subject subject);

    List<Subject> getAllSubjects();

    Subject getSubjectById(Long id);
}
